src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -143,6 +151,16 @@ internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToCo
                     return success;
                 }
 
+                if (state.Current.ParentProperty?.TryPopulate(ref state) == true)
```

Comment:
As you probably already found out, this method logic is... delicate :-) As with the previous and next blocks, could you add a comment explaining why exactly this step is necessary _after_ polymorphic dispatch and why it needs to happen _before_ the subsequent "Handle metadata" stage?

Reply:
actually the order doesn't matter because we throw with any kind of metadata on `Configure` (both polymorphic deserialization and reference handling)

Reply:
comment will be included in the next iteration